jetty-client/src/main/java/org/eclipse/jetty/client/AbstractConnectionPool.java

```diff
@@ -44,13 +44,13 @@
      * The bottom 32 bits represent the total connections and the top 32 bits represent the pending connections.
      */
     private final AtomicBiInteger connections = new AtomicBiInteger();
-    private final Destination destination;
+    private final HttpDestination destination;
     private final int maxConnections;
     private final Callback requester;
 
     protected AbstractConnectionPool(Destination destination, int maxConnections, Callback requester)
     {
-        this.destination = destination;
+        this.destination = (HttpDestination)destination;
```

Comment:
IF this can now only be a HttpDestination, why not change the signature of the constructor?

Reply:
Because people may have created their own subclasses and it would break them.
Will do the change in 10 though.

Reply:
You could create a second constructor that does the right thing and deprecate the old one (preparing people for the Jetty 10 change).

Reply:
@joakime good point, done.